File: 2020/day-two/daytwo.py
with open('input.txt') as f:
    l = f.read().splitlines()

# question  1
total = 0
for line in l:
    parts = line.split()
    lower = int(parts[0].split('-')[0])
    upper = int(parts[0].split('-')[1])
    letter = parts[1][0]
    st = parts[2]
    letter = st.count(letter)
    if letter < lower or letter > upper:
        total+= 1

print(1000-total)



# question 2 
total = 0
for line in l:
    parts = line.split()
    first_position = int(parts[0].split('-')[0])
    second_position = int(parts[0].split('-')[1])
    letter = parts[1][0]
    st = parts[2]

    cond_one = st[first_position-1] == letter
    cond_two = st[second_position-1] == letter
    if cond_one + cond_two == 1:
        total+=1
print(total)

File: 2020/day-two/daytwo.R
library(tidyverse)

l <- readLines("input.txt") #readLines 
    
head(l)

# part 1


# l %>% 
#     separate(V1, into = c("bounds", "letter", "pattern"), sep = " ") %>% 
#     separate(bounds, into = c("lower", "upper"), sep = "-") %>%
#     mutate(letter = gsub(":", "", letter), 
#     count = str_count(pattern, letter), 
#     good = if_else((count >= lower & count <= upper), 1, 0)) %>% 
#     count(good)

check <- function(x) {
  numbers <- as.numeric(str_extract_all(x, "[0-9]+")[[1]])
  string <- str_extract_all(x, "[a-z]+")[[1]]
  
  count <- str_count(string[2], string[1])
  (numbers[1] <= count) & (numbers[2] >= count) 
}
sum(map_lgl(l, check))


# part 2 

check_2 <- function(x) {
  numbers <- as.numeric(str_extract_all(x, "[0-9]+")[[1]])
  chars <- str_extract_all(x, "[a-z]+")[[1]]
  
  loc1 <- str_sub(chars[2], numbers[1], numbers[1])
  loc2 <- str_sub(chars[2], numbers[2], numbers[2])
     
  sum(loc1 == chars[1], loc2 == chars[1]) == 1
  
}
sum(map_lgl(l, check_2))




